### PR TITLE
Atlas Users Service

### PR DESCRIPTION
**Pull request by marinsalinas**
Add Atlas Users Service to perform the following requests:

1. Get all users from the project.
2. Get by ID
3. Get by Username
4. Create Atlas Users.

> **NOTE**: We are pending with Update and Delete Atlas User since these calls are not enabled in the public API.

### CODE REVIEW

File: mongodbatlas/atlas_users.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	LastName     string      `json:"lastName"`
+	Roles        []AtlasRole `json:"roles"`
+	TeamIds      []string    `json:"teamIds,omitempty"`
+	Username     string      `json:"username"`
+	MobileNumber string      `json:"mobileNumber"`
+	Password     string      `json:"password"`
+	Country      string      `json:"country"`
+}
+
+//GetAllAtlasUsers gets all users.
```
Comment (PacoDw) [inline mongodbatlas/atlas_users.go]: `GetAllAtlasUser` comment name doesn't refer to the function name

File: mongodbatlas/atlas_users.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	root := new(AtlasUser)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+//Get gets a single atlas user by name.
```
Comment (PacoDw) [inline mongodbatlas/atlas_users.go]: The same above

File: mongodbatlas/clusters.go [hunk 3]
```diff
@@ -157,7 +157,7 @@ var DefaultDiskSizeGB map[string]map[string]float64 = map[string]map[string]floa
 		"M200": 1500,
 		"M300": 2200,
 	},
-	"AZURE": map[string]float64{
+	"AZURE": {
 		"M10":  32,
 		"M20":  32,
 		"M30":  32,
```
Comment (PacoDw) [inline mongodbatlas/clusters.go]: Very nice changes 👍 🥇

File: mongodbatlas/teams.go [hunk 4]
```diff
@@ -55,16 +55,6 @@ type AtlasUserAssigned struct {
 	TotalCount int         `json:"totalCount"`
```
[the remark this answers is not in the document]
Comment (marinsalinas) [reply mongodbatlas/teams.go]: I made this change because I wanted to reuse this struct in both services.

[the remark this answers is not in the document]
Comment (themantissa) [reply mongodbatlas/teams.go]: Ah! Okay that makes sense.  Thank you!

**Pull request merged**